packages/babel-plugin-proposal-pipeline-operator/test/fixtures/smart/for-update/output.js

```diff
@@ -0,0 +1,9 @@
+let sum = 0;
+
+for (var i = 0; i <= 10; i = (_i = i, _i + 1)) {
+  var _i;
```

Comment:
I just noticed that this `_i` is in the wrong scope, but since it is a bug with `@babel/traverse` I won't block this PR.
I'll work on it in the meantime.

Reply:
Oh, `var`s are hoisted and `path.scope.push` uses `var`, so even if it is inside the loop body it is ok.